src/distinct.rs

```diff
@@ -268,13 +287,20 @@ impl DistanceResult {
         self.mean_closest_distance = 0.0;
         self.min_closest_distance = scalar::MAX;
 
+        let mut closest_pair_set = false;
+
         for (i, (dist, closest_i)) in self.closest_distances.iter().enumerate() {
             self.mean_closest_distance += *dist;
 
-            if *dist < self.min_closest_distance {
-                self.min_closest_distance = *dist;
+            // the closest pair must ignore the fixed_colors because we cannot change them
```

Comment:
what if the closest pair has one fixed color and one non-fixed color? Wouldn't we still want to change the non-fixed one in this case?

Reply:
good catch! you're absolutely right. I should have fixed now by forcing the process to use the non fixed color in case one of the two is fixed.